### PR TITLE
Cleanup singleton part2

### PR DESCRIPTION
**Pull request by Nicogene**
This PR c11-fy `Bottle`, `BottleImpl` and `MultiNameSpace` singletons.

demoRedball tested successfully on the simulator.

### CODE REVIEW

**Comment by drdanz** [comment]
Added other c++11 singleton-related stuff as discussed with @Nicogene

File: src/libYARP_OS/src/Network.cpp [hunk 1]
```diff
@@ -1299,14 +1290,20 @@ std::string NetworkBase::getPathSeparator() {
 #endif
 }
 
+namespace {
+static std::mutex& getNetworkMutex()
```
Comment (Nicogene) [inline src/libYARP_OS/src/Network.cpp]: ❤️

File: src/libYARP_OS/src/Network.cpp [hunk 2]
```diff
@@ -1299,14 +1290,20 @@ std::string NetworkBase::getPathSeparator() {
...
+{
+    static std::mutex mutex;
+    return mutex;
+}
+} // namespace
+
 void NetworkBase::lock() {
-    ThreadImpl::init();
-    ThreadImpl::threadMutex->wait();
+    getNetworkMutex().lock();
```
Comment (Nicogene) [inline src/libYARP_OS/src/Network.cpp]: ❤️

File: src/libYARP_OS/src/Network.cpp [hunk 3]
```diff
@@ -1299,14 +1290,20 @@ std::string NetworkBase::getPathSeparator() {
...
 }
 
 void NetworkBase::unlock() {
-    ThreadImpl::init();
-    ThreadImpl::threadMutex->post();
+    getNetworkMutex().unlock();
```
Comment (Nicogene) [inline src/libYARP_OS/src/Network.cpp]: ❤️

File: doc/yarp_env_variables.md [hunk 4]
```diff
@@ -93,6 +92,7 @@ Deprecated Environmental Variables
 | Environmental variable        | Description | Related documentation page |
 |:-----------------------------:|:-----------:|:--------------------------:|
 | `YARP_POLICY`                 |  Legacy variable used to change the behavior of the ResourceFinder. Deprecated since YARP 2.3.65  | |
+| `YARP_STACK_SIZE`             | Default stack size (in bytes) for YARP threads. Deprecated since YARP 3.0.0 | |
```
Comment (traversaro) [inline doc/yarp_env_variables.md]: Deprecated or removed? If it is removed, I think we can remove it from the table.

Comment (Nicogene) [reply doc/yarp_env_variables.md]: :+1:  removing it later in a commit of cleanup of the documentation

Comment (drdanz) [reply doc/yarp_env_variables.md]: Removed, because it is no longer an available option with c++11 threads

**Pull request merged**